File: app/src/main/java/com/geowatershed/app/data/model/PhotoSet.kt
package com.geowatershed.app.data.model

enum class PhotoSet(val note: String) {
    Before("PRE-WORK PHOTO"),
    After("POST-WORK PHOTO"),
    ;

    companion object {
        /** Reads a photo set from a stored string without throwing. */
        fun parse(raw: String?): PhotoSet = entries.firstOrNull { it.name == raw } ?: Before
    }
}

File: geowatershed_flutter/lib/models/photo_set.dart
/// Which half of a before/after evidence pair a photo belongs to.
enum PhotoSet {
  before('PRE-WORK PHOTO'),
  after('POST-WORK PHOTO');

  const PhotoSet(this.note);

  final String note;

  static PhotoSet parse(String? raw) {
    if (raw == null) return before;
    final key = raw.toLowerCase();
    return PhotoSet.values.firstWhere((p) => p.name.toLowerCase() == key, orElse: () => before);
  }
}
